### PR TITLE
fix: feature branch for auto-coding issues

### PR DESCRIPTION
**Pull request by Z3RO-O**
## Clickup
https://app.clickup.com/t/86cwh4z32

## Code Coverage
Unit Test Coverage | % | values
-- | -- | --
Statements | 95.86% | ( 19103 / 19928 )
Branches | 91.13% | ( 10485 / 11505 )
Functions | 94.26% | ( 5701 / 6048 )
Lines | 95.89% | ( 18233 / 19014 )

## UI Preview
1. Added auto-coded icon in add-edit-expenses
image
 
2. Added auto-coded icon in view expense
image

3. Scanning loader  
image

## Summary by CodeRabbit

- **New Features**
	- Introduced an input property for auto-coded data in the currency component, enhancing user feedback on currency and amount coding status.
	- Added a message display for auto-coded status, improving clarity for users during transactions.

- **Usability Improvements**
	- Enhanced feedback mechanisms to indicate whether currency and amount have been auto-coded, improving user interaction and reducing data entry errors.
	- Updated lifecycle methods to ensure timely updates to user feedback based on form values.

### CODE REVIEW

File: src/app/shared/components/fy-currency/fy-currency.component.spec.ts [hunk 1]
```diff
@@ -200,6 +200,8 @@
 
     // Test case 2: inner value with same currency
     const innerValue2 = {
+      orig_amount: null,
+      orig_currency: null,
```
[the remark this answers is not in the document]
Comment (Z3RO-O) [reply src/app/shared/components/fy-currency/fy-currency.component.spec.ts:204]: yes

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -1738,7 +1743,14 @@
...
 
-    from(this.loaderService.showLoader('Loading expense...', 15000))
+    // To conditionally change the loader for add and edit expense
+    const loader$ = this.activatedRoute.snapshot.params.dataUrl
+      ? from(
+          this.loaderService.showLoader('Scanning information from the receipt...', 15000, 'assets/images/scanning.gif')
+        )
+      : from(this.loaderService.showLoader('Loading expense...', 15000));
+
+    loader$
```
[the remark this answers is not in the document]
Comment (Z3RO-O) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:1753]: Should this be changed like this?
@Dimple16

[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:1753]: This is fine

**Review by suyashpatil78 — APPROVED** [review]

File: src/app/shared/components/fy-currency/fy-currency.component.spec.ts [hunk 3]
```diff
@@ -266,12 +269,10 @@ describe('FyCurrencyComponent', () => {
...
-    modalController.create.and.returnValue(
-      Promise.resolve({
-        present: () => {},
-        onWillDismiss: () => Promise.resolve({ data: { amount: 100, homeCurrencyAmount: 500 } }),
-      } as any)
-    );
+    modalController.create.and.resolveTo({
+      present: () => {},
+      onWillDismiss: () => Promise.resolve({ data: { amount: 100, homeCurrencyAmount: 500 } }),
+    } as any);
```
[the remark this answers is not in the document]
Comment (Z3RO-O) [reply src/app/shared/components/fy-currency/fy-currency.component.spec.ts:275]: removed any

**Review by devendrafyle — APPROVED** [review]

**Pull request merged**